### PR TITLE
Add batching to stackdriver adapter

### PR DESCRIPTION
**Pull request by bianpengyuan**
Add time series batching logic and size limit to Stackdriver adapter. cc @quentinmit

### CODE REVIEW

File: mixer/adapter/stackdriver/metric/bufferedClient.go [hunk 1]
```diff
@@ -74,19 +88,23 @@ func (b *buffered) Send() {
...
-	if err != nil {
-		_ = b.l.Errorf("Stackdriver returned: %v\nGiven data: %v", err, merged)
-	} else {
-		b.l.Debugf("Successfully sent data to Stackdriver.")
+	batches := batchTimeSeries(merged, b.timeSeriesBatchSize)
+
+	for _, timeSeries := range batches {
+		err := b.pushMetrics(context.Background(),
+			&monitoringpb.CreateTimeSeriesRequest{
+				Name:       fmt.Sprintf("projects/%s", b.project),
```
Comment (geeknoid) [inline mixer/adapter/stackdriver/metric/bufferedClient.go]: Instead of Sprintf, you can just use concatenation which is considerably more efficient.

**Comment by bianpengyuan** [comment]
@douglas-reid verified in my set up that the fix eliminates 200 limit error. removed the TODO line.

I cannot find the reference about per minute limit... @quentinmit except this 200 batch size limit, is there any other limitation that you are aware of about CreateTimeSeries API?

File: mixer/adapter/stackdriver/metric/bufferedClient.go (hunk 1 above)
Comment (bianpengyuan) [reply mixer/adapter/stackdriver/metric/bufferedClient.go]: Thanks for reviewing! Done.

**Comment by quentinmit** [comment]
@bianpengyuan There are several other quota limits... there is a default limit of 6kqpm (~=100qps) calls to CreateTimeSeries, as well as a limit of 100k "active time series" (# of time series with any data in the last 24h). There are a bunch more limits as well at: https://cloud.google.com/monitoring/quotas

Customers can either self-raise or file a ticket to raise any of these quotas if they hit them, so we shouldn't bake any of them into Istio. Just know that you will get a 429 error back if you exceed any of the quotas.

**Comment by douglas-reid** [comment]
@hklai or @geeknoid can we please merge this PR?

**Comment by hklai** [comment]
@douglas-reid  could you comment on the risk and why this is needed in 1.0?

**Comment by douglas-reid** [comment]
@hklai this poses little to no risk for Istio in general. for SD users, this PR prevents loss of metrics. some users have already reported experiencing this issue.

**Pull request merged**